templates/cc/string.go

```diff
@@ -135,12 +144,39 @@ const strTpl = `
 		}
 		*/}}
 	{{ else if $r.GetHostname }}
-		{{ unimplemented }}
-		{{/* TODO(akonradi) implement hostname constraints
-		if err := m._validateHostname({{ accessor . }}); err != nil {
-			return {{ errCause . "err" "value must be a valid hostname" }}
+	{
+		// Take a copy so we can lowercase it.
+		std::string value = {{ accessor . }};
+		std::transform(value.begin(), value.end(), value.begin(), [](unsigned char c){ return std::tolower(c); });
+
+		if (value.length() > 253) {
+			{{ err . "value must be a hostname, and hostname cannot exceed 253 characters" }}
 		}
-		*/}}
+
+		const std::regex dot_regex{"\\."};
+		const std::vector<std::string> domain_parts{
```

Comment:
No need to construct a vector here, just iterate directly over the `std::sregex_token_iterator` with a `for` or `while` loop. Otherwise this is just more redundant copying and allocation.

Reply:
Ah yeah good catch :+1: